### PR TITLE
feat: Add pagination support serializer mixin

### PR DESCRIPTION
**Pull request by bertdeblock**
Adds pagination support to a serializer. It adds a `pagination` object to `meta` based on `links` and `meta` provided by the API. This way, `result.meta.pagination` becomes accessible inside templates to render a pagination component.

### CODE REVIEW

File: tests/dummy/config/optional-features.json [hunk 1]
```diff
@@ -1,3 +1,3 @@
 {
-  "jquery-integration": false
+  "jquery-integration": true
```
Comment (bertdeblock) [inline tests/dummy/config/optional-features.json]: ED needs jQuery and this is only needed for testing purposes.

File: tests/unit/serializers/json-api-bagaaravel-test.js [hunk 2]
```diff
@@ -44,4 +44,40 @@ module('Unit | Serializer | json api bagaaravel', function (hooks) {
...
     assert.notOk(existingUserSerialized.data.relationships);
   });
+
+  test('normalizeQueryResponse', async function (assert) {
+    let store = this.owner.lookup('service:store');
+    let Adapter = this.owner.resolveRegistration('adapter:application');
+
+    Adapter.reopen({
+      query() {
+        return Promise.resolve({
```
Comment (bertdeblock) [inline tests/unit/serializers/json-api-bagaaravel-test.js]: Simulate a query response.

File: tests/unit/mixins/pagination-support-serializer-test.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            perPage: 15,
+            total: 125,
+          },
+        });
+      },
+    });
+
+    let Serializer = JSONAPIBagaaravelSerializer.extend(PaginationSupportSerializerMixin);
+
+    this.owner.register('adapter:application', Adapter);
```
Comment (bertdeblock) [inline tests/unit/mixins/pagination-support-serializer-test.js]: You could also generate an application adapter in our dummy app and that would automagically be used in our tests. You could use `reopen` to adapt behaviour per test. But this approach seemed much cleaner/explicit though it's a bit more verbose.

Comment (Windvis) [reply tests/unit/mixins/pagination-support-serializer-test.js]: I think this is a way better solution anyway. This code is very specific for this test. It shouldn't be added to the serializer that will be used by other tests. Good job!

**Pull request merged**